test/factories/users.rb

```diff
@@ -71,13 +71,15 @@
     expiry_date_for_2sv_exemption { (Time.zone.today + 1).to_date }
   end
 
-  factory :user_with_pending_email_change, parent: :user do
+  trait :with_pending_email_change do
```

Comment:
I think it might be better to nest this, and the `invited_user` trait, inside the `:user` factory rather than making them top-level. Although they shouldn't cause any problem being at the top-level I think there's an implication that they're useful outside of the `user` factories and I'm not sure that's true in these cases.

I only discovered that top-level traits were possible when they broke in a recent upgrade of factory_bot - ee01a1b318aed3a961c8660f7d0a2110c32d66ac.

Reply:
I'd completely missed that - good catch!

Reply:
As we just discussed in Slack, I need these traits to be available to the `api_user` factory and not just the `user` factory and so without introducing some kind of base class, I can't see how to do this. Marking as resolved.